app/Http/Controllers/API/UserController.php

```diff
@@ -0,0 +1,273 @@
+
+            $returnData = [];
+
+            $returnData['id'] = $user->id;
+            ($includeEmail) ? $returnData['email'] = $user->email : null;
+            ($includeName) ? $returnData['first_name'] = $user->first_name : null;
+            ($includeName) ? $returnData['last_name'] = $user->last_name : null;
+            $returnData['rating'] = $user->rating_short;
+            ($includeDivisions) ? $returnData['region'] = $user->region : null;
+            ($includeDivisions) ? $returnData['division'] = $user->division : null;
+            ($includeDivisions) ? $returnData['subdivision'] = $user->subdivision : null;
+            $returnData['atc_active'] = boolval($user->atc_active);
+            ($includeEndorsements) ? $returnData['endorsements'] = $user->endorsements : null;
+            ($includeRoles) ? $returnData['roles'] = $user->roles : null;
+            ($includeTraining) ? $returnData['training'] = $user->training : null;
+
+            return $returnData;
+        });
+    }
+
+    /**
+     * Map out the endorsements and put them in a category for each type
+     *
+     * @return array
+     */
+    private function mapEndorsements(Collection $endorsements)
+    {
+
+        $returnData = [
+            'visiting' => [],
+            'examiner' => [],
+            'training' => [
+                'solo' => [],
+                's1' => [],
+            ],
+            'masc' => [],
+        ];
+
+        // Remembner adding the related rating/area to the query
+        foreach ($endorsements as $endorsement) {
+
+            switch ($endorsement->type) {
+                case 'VISITING':
+                    array_push($returnData['visiting'], $this->mapEndorsementDetails($endorsement, $endorsement->type));
+                    break;
+                case 'EXAMINER':
+                    array_push($returnData['examiner'], $this->mapEndorsementDetails($endorsement, $endorsement->type));
+                    break;
+                case 'SOLO':
+                    array_push($returnData['training']['solo'], $this->mapEndorsementDetails($endorsement, $endorsement->type));
+                    break;
+                case 'S1':
+                    array_push($returnData['training']['s1'], $this->mapEndorsementDetails($endorsement, $endorsement->type));
+                    break;
+                case 'MASC':
+                    array_push($returnData['masc'], $this->mapEndorsementDetails($endorsement, $endorsement->type));
+                    break;
+            }
+
+        }
+
+        empty($returnData['visiting']) ? $returnData['visiting'] = null : null;
+        empty($returnData['examiner']) ? $returnData['examiner'] = null : null;
+        empty($returnData['training']['solo']) ? $returnData['training']['solo'] = null : null;
+        empty($returnData['training']['s1']) ? $returnData['training']['s1'] = null : null;
+        empty($returnData['masc']) ? $returnData['masc'] = null : null;
```

Comment:
You can do either, but I'm curious why you specifically want to return null instead of an empty array.

Reply:
Because an empty array would make a response like
`{parameter: [], parameter2: []}` while I want a more clear response that it's not existing by just giving null instead ´{parameter: null, parameter2: null}. Then when you code stuff for the endpoint you can ask `if($parameter)` instead of `if($parameter && $paramter->isNotNull())` or `isNotEmpty()`